backend/models/database.py

```
@@ -11,67 +11,120 @@
+        self._data[doc_id] = doc
+        return type('Result', (), {'inserted_id': doc_id})()
+
+    def find_one(self, query, projection=None):
+        for doc in self._data.values():
+            if all(doc.get(k) == v for k, v in query.items()):
+                if projection:
+                    return {k: doc.get(k) for k in projection if k in doc} if projection else doc
+                return doc
+        return None
+
+    def update_one(self, query, update, upsert=False):
```

Ruff (0.14.13)

Unused method argument: `upsert`

(ARG002)